agent/integration/test_helpers.go

```diff
@@ -229,6 +229,13 @@ func (t *testAgentEndpoint) server(extraRoutes ...route) *httptest.Server {
 			t.calls[req.URL.Path] = append(t.calls[req.URL.Path], b)
 			t.mtx.Unlock()
 
+			// We also require job tokens are used for authentication
+			if !strings.HasPrefix(req.Header["Authorization"][0], "Token bkaj_") {
+				fmt.Printf("Invalid authorization header, expected job token, got: %q\n", req.Header["Authorization"][0])
+				rw.WriteHeader(http.StatusUnauthorized)
```

Comment:
Suggestion 1: pull the authz header out of `req.Header` with the `Get` method, which should handle missing headers and canonicalisation

Suggestion 2: I'd serve the error message as the HTTP response body, and prefer stderr to stdout - `http.Error` and `fmt.Fprintf(os.Stderr, ...)` or `log.Printf` functions can help

Suggestion 3: some [tweaks](https://google.github.io/styleguide/go/decisions#got-before-want) to the message formatting

```suggestion
			authzHeader := req.Header.Get("Authorization")
			if !strings.HasPrefix(authzHeader, "Token bkaj_") {
				msg := fmt.Sprintf("Authorization header = %q, want job token prefix 'Token bkaj_'", authzHeader)
				fmt.Fprintln(os.Stderr, msg)
				http.Error(w, msg, http.StatusUnauthorized)
```